mteb/tasks/Clustering/eng/ArxivClusteringP2PFast.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import annotations
+
+import datasets
+
+from mteb.abstasks.AbsTaskClusteringFast import AbsTaskClusteringFast
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+
+class ArxivClusteringP2PFast(AbsTaskClusteringFast):
+    metadata = TaskMetadata(
+        name="ArxivClusteringP2PFast",
+        description="Clustering of titles+abstract from arxiv. Clustering of 30 sets, either on the main or secondary category",
+        reference="https://www.kaggle.com/Cornell-University/arxiv",
+        dataset={
+            "path": "mteb/arxiv-clustering-p2p",
+            "revision": "a122ad7f3f0291bf49cc6f4d32aa80929df69d5d",
+        },
+        type="Clustering",
+        category="p2p",
+        eval_splits=["test"],
+        eval_langs=["eng-Latn"],
+        main_score="v_measure",
+        date=None,
+        form=None,
+        domains=None,
+        task_subtypes=None,
+        license=None,
+        socioeconomic_status=None,
+        annotations_creators=None,
+        dialect=None,
+        text_creation=None,
+        bibtex_citation=None,
+        n_samples={"test": 732723},
+        avg_character_length={"test": 1009.98},
+    )
+
+    def dataset_transform(self):
+        sent = self.dataset["test"]["sentences"][0]
+        lab = self.dataset["test"]["labels"][0]
+
+        self.dataset["test"] = datasets.Dataset.from_dict(
+            {"sentences": sent, "labels": lab}
+        )
```

Comment:
now it is a dataset with two columns sentences and labels list[str], before it was a dataset of two column (list[list[str]]), where the first list is a cluster and the second list is the labels/sentences of that cluster.

Reply:
One frustrating part (for backward compatiblity) is that the clusters are not from the same distribution.

E.g. for Swedn we had 2 clusters articles and 2 clusters of summaries (and a bug, but not relevant here). Which naturally isn't the same as sampling from articles and summaries and generating clusters. The same is the case for ArxivClusteringP2P (but I am unsure if this a general case or not - my gut would say that about ~30% has exceptions like these).

Reply:
> This change is to make it simpler?

Yep as the user-defined clusters (sets of documents) are no longer needed, which allows us to sample clusters with some overlap (allowing for a notably larger sample) 

> In that Swedn case, shouldn't it be two separate datasets then, maybe SwednP2P & SwednS2S Clustering?

Yea that might indeed be a better solution

> I think for most current datasets like arxiv etc all clusters are from the same dist, no?

I think you are right, though arxivClusteringP2P seems to be from multiple dists (as I understand the paper at least)

Reply:
I created a new arxiv P2P dataset, let me know if there is issue with it.